src/parser/mod.rs

```diff
@@ -3744,10 +3744,18 @@ impl<'a> Parser<'a> {
         self.expect_token(&Token::LParen)?;
         let in_op = if self.parse_keyword(Keyword::SELECT) || self.parse_keyword(Keyword::WITH) {
             self.prev_token();
-            Expr::InSubquery {
-                expr: Box::new(expr),
-                subquery: self.parse_query()?,
-                negated,
+            if self.dialect.supports_in_query_body() {
+                Expr::InSubquery {
+                    expr: Box::new(expr),
+                    subquery: self.parse_query_body(self.dialect.prec_unknown())?,
+                    negated,
+                }
+            } else {
+                Expr::InSubquery {
+                    expr: Box::new(expr),
+                    subquery: Box::new(SetExpr::Query(self.parse_query()?)),
+                    negated,
+                }
             }
```

Comment:
would it make sense to skip the `supports_in_query_body` flag entirely? thinking since the new representation `SetExpr` is a superset of the old query

Reply:
I am not sure since that would allow parsing this syntax on dialects that don't actually support it, right?

Reply:
Ah yeah I think that would be fine to have the parser always accept the syntax

Reply:
Okay, done, and fixed to work in all cases!